### PR TITLE
Continent Implementation

### PR DESCRIPTION
**Pull request by RootCero**
Added the ability to specify the continent in Whitelist and Blacklist.

### CODE REVIEW

File: Configuration.cpp [hunk 1]
```diff
@@ -60,6 +60,33 @@ bool Configuration::readConfigFile(char *path) {
...
+    } else {
+      for(Json::Value::ArrayIndex i = 0; i != root["continents"]["blacklist"].size(); i++) {
+	std::string continent = root["continents"]["blacklist"][i].asString();
+
+	trace->traceEvent(TRACE_INFO, "Adding %s to continents blacklist", continent.c_str());
+	continents[country2u16((char*)continent.c_str())] = MARKER_DROP;
+      }
+    }
+  }
+
```
Comment (lucaderi) [inline Configuration.cpp:89]: Can we refactor the code in order to merge continent and country into the same parsing logic?

File: Configuration.cpp [hunk 2]
```diff
@@ -138,4 +165,16 @@ Marker Configuration::getCountryMarker(char *country) {
...
+{
+  u_int16_t id = country2u16(continent);
+  std::unordered_map<u_int16_t, Marker>::iterator it = continents.find(id);
+
+  if(it == continents.end())
+    return(default_marker); /* Not found */
+  else
+    return(it->second);
+
+  //if(strcmp())
```
Comment (lucaderi) [inline Configuration.cpp:178]: Remove

File: Configuration.cpp [hunk 3]
```diff
@@ -138,4 +165,16 @@ Marker Configuration::getCountryMarker(char *country) {
     return(it->second);
 }
 
+Marker Configuration::getContinentMarker(char *continent)
```
Comment (lucaderi) [inline Configuration.cpp:168]: Refactor code here too

File: NwInterface.cpp [hunk 4]
```diff
@@ -226,15 +226,17 @@ Marker NwInterface::makeVerdict(u_int8_t proto, u_int16_t vlanId,
...
   const char *proto_name = getProtoName(proto);
   bool pass_local = true;
   Marker m, src_maker, dst_marker;
   
+  //convert values between host and network byte order
   sport = ntohs(sport), dport = ntohs(dport);
 
   // trace->traceEvent(TRACE_NORMAL, "Processing %u -> %u", sport, dport);
   
+  //Valuta se il protocollo è monitorato
```
Comment (lucaderi) [inline NwInterface.cpp:239]: Remove non-English comment

File: NwInterface.cpp [hunk 5]
```diff
@@ -255,15 +257,22 @@ Marker NwInterface::makeVerdict(u_int8_t proto, u_int16_t vlanId,
     break;
   }
 
+  //Prende il default marker (specificato nel JSON)
```
Comment (lucaderi) [inline NwInterface.cpp:260]: Remove non-English comment

File: NwInterface.cpp [hunk 6]
```diff
@@ -255,15 +257,22 @@ Marker NwInterface::makeVerdict(u_int8_t proto, u_int16_t vlanId,
...
+    in  network  byte  order,  to a string in IPv4 dotted-decimal notation.
+    The string is returned in a statically allocated buffer,  which  subse‐
+    quent calls will overwrite.*/
   host = inet_ntoa(in);
   strncpy(src_host, host, sizeof(src_host)-1);
-
-  if(geoip->lookup(host, country_code, sizeof(country_code), NULL, 0)) {
-    src_maker = conf->getCountryMarker(country_code);
+  if(geoip->lookup(host, country_code, sizeof(country_code), countinent_code, sizeof(countinent_code))) {
+    printf("\n=== %s ===\n", countinent_code);
```
Comment (lucaderi) [inline NwInterface.cpp:271]: Remove debug

File: NwInterface.cpp [hunk 7]
```diff
@@ -276,9 +285,12 @@ Marker NwInterface::makeVerdict(u_int8_t proto, u_int16_t vlanId,
   host = inet_ntoa(in);
   strncpy(dst_host, host, sizeof(dst_host)-1);
 
-  if(geoip->lookup(host = inet_ntoa(in), country_code, sizeof(country_code), NULL, 0)) {
+  if(geoip->lookup(host = inet_ntoa(in), country_code, sizeof(country_code), countinent_code, sizeof(countinent_code))) {
+    printf("\n=== %s ===\n", countinent_code);
```
Comment (lucaderi) [inline NwInterface.cpp:289]: Remove debug

File: sample_config.json [hunk 8]
```diff
@@ -1,13 +1,17 @@
 {
     "queue_id": 0,
-    "default_marker": "DROP",
+    "default_marker": "Drop",
```
Comment (lucaderi) [inline sample_config.json:3]: We need to handle case-insensistive comparisons

File: sample_config.json [hunk 9]
```diff
@@ -1,13 +1,17 @@
...
+    "continents": {
+		"whitelist": ["Nor","Eur"],
```
Comment (lucaderi) [inline sample_config.json:5]: Can you please write somewhere what are the continent acronyms supported?

**Review by lucaderi — CHANGES_REQUESTED** [review]
This code is a step forward but there are some issues to be addressed before merge

File: Configuration.cpp (hunk 1 above)
Comment (lucaferret) [reply Configuration.cpp:89]: We will resolve in the pull of Thursday

**Comment by lucaderi** [comment]
Closing PR as of https://github.com/ntop/ipt_geofence/pull/11

**Pull request closed without merging**